utils.go

```diff
@@ -141,6 +141,15 @@ func toFile(file string, s *state) {
 	}
 }
 
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
```

Comment:
No super important here because these slices will always be small but it would be better to use something like `sort.SearchStrings` instead of a linear search.

Reply:
Yep that's possible but for me given that here's no performance implication, this one is even harder to read:

```go

sort.Sort(StringSlice(s.PreconfiguredHelmRepos))

possible_index = sort.SearchStrings(s.PreconfiguredHelmRepos, repoOrDir)
isRepo = isRepo || (possible_index < len(s.PreconfiguredHelmRepos) && s.PreconfiguredHelmRepos[possible_index] == repoOrDir)
```
The other thing that can happen is to use a map[string]bool type, and instead of the PreconfiguredHelmRepos have something like:

```yaml
manageHelmRepos:
  my-repo: false
```

This would mean that:
1) by default all helm repos are managed that does not have this key
2) if the user sets it to true then those repos are still managed
3) if it's set to false then it's an opt out

But I would prefer to go with this changeset for now, unless you insist on me implementing this approach.